src/core/src/sim/mc/moves/rotate.rs

```diff
@@ -114,7 +116,9 @@ impl MCMove for Rotate {
 
     fn update_amplitude(&mut self, scaling_factor: Option<f64>) {
         if let Some(s) = scaling_factor {
-            self.theta *= s;
+            if (s * self.theta).to_degrees() <= 180.0 {
+                self.theta *= s;
+            }
```

Comment:
Maybe we can send a warning (using `warn_once!`) if the move amplitude would have gotten bigger than 180°.